src/main/java/edu/virginia/lib/indexing/ASpaceObject.java

```diff
@@ -515,4 +523,100 @@ public void printOutRawData() {
+		MarcFactory factory = MarcFactory.newInstance();
+		Record r = factory.newRecord();
+		DataField df;
+		Subfield sf;
+		
+		String title = record.getString("title");
+        char nonIndexChars = '0';
+        if (title.startsWith("A "))
+        	nonIndexChars = '2';
+        else if (title.startsWith("The "))
+        	nonIndexChars = '4';
+        
+        df = factory.newDataField("245", '0', nonIndexChars);
+		sf = factory.newSubfield('a', title);
+		df.addSubfield(sf);
+		r.addVariableField(df);
+		
+		df = factory.newDataField("590", '1', ' ');
+		sf = factory.newSubfield('a', "From ArchivesSpace: " + record.getString("record_uri"));
+		df.addSubfield(sf);
+		r.addVariableField(df);
+		
+		
+		
+		//pull desired resource from ArchivesSpace, get reference for each unique top_container
+        HashSet<String> topContainers = new HashSet<String>();
+        JsonArray children = record.getJsonArray("children");
+        
+        for (JsonValue child : children) {
+        	if (((JsonObject)child).getString("node_type").equals("archival_object")) {
+        		ASpaceObject temp;
+        		if (this instanceof ASpaceCollection)
+        			temp = new ASpaceCollection(c, ((JsonObject)child).getString("record_uri"));
+        		else
+        			temp = new ASpaceAccession(c, ((JsonObject)child).getString("record_uri"));
+        		
+        		JsonArray instances = temp.record.getJsonArray("instances");
+        		
+        		for (JsonValue instance : instances) {
+        			JsonObject subContainer = ((JsonObject)instance).getJsonObject("sub_container");
+        			topContainers.add(subContainer.getJsonObject("top_container").getString("ref"));
+        		}
+        	}
+        }
+        
+        
+        
+        //generate a 999 field for each top_container
+        for (String s : topContainers) {
+        	ASpaceObject topContainer;
+        	if (this instanceof ASpaceCollection)
+        		topContainer = new ASpaceCollection(c, s);
+        	else
+        		topContainer = new ASpaceAccession(c, s);
+        	
+        	df = factory.newDataField("999", ' ', ' ');
+    		
+    		String barcode;
+    		if (topContainer.record.get("barcode") != null) {
+    			barcode = topContainer.record.getString("barcode");
+        	} else {
+    			String uri = topContainer.record.getString("uri");
+    			String topContainerNumber = uri.substring(uri.lastIndexOf("/") + 1);
+    			String repository = uri.split("/")[2];
+    			barcode = "AS:R" + repository + "C" + topContainerNumber;
+    		}
+    		sf = factory.newSubfield('i', barcode);
+    		df.addSubfield(sf);
+    		
+    		sf = factory.newSubfield('m', this.getLibrary(topContainer.record));
+    		df.addSubfield(sf);
+    		
+    		r.addVariableField(df);
+        }
+        
+        
+        
+        //write marc object to file
+  		try {
+  			if (!file.exists())
+  				file.createNewFile();
+  			
+  			FileOutputStream o = new FileOutputStream(file);
```

Comment:
This stream won't be closed in the event of an exception.  There's an concise way to do this starting with java 7:
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html